packages/fxa-auth-server/lib/routes/emails.js

```diff
@@ -696,14 +699,23 @@ module.exports = (
                 Date.now() - secondaryEmailRecord.createdAt;
               const minUnverifiedAccountTime =
                 config.secondaryEmail.minUnverifiedAccountTime;
-              if (msSinceCreated >= minUnverifiedAccountTime) {
+              const emailNewlyCreated =
+                msSinceCreated >= minUnverifiedAccountTime;
+              if (
+                emailNewlyCreated &&
+                !(await stripeHelper.hasActiveSubscription(
+                  secondaryEmailRecord.uid
+                ))
+              ) {
                 await db.deleteAccount(secondaryEmailRecord);
                 log.info('accountDeleted.unverifiedSecondaryEmail', {
                   ...secondaryEmailRecord,
                 });
                 return;
-              } else {
+              } else if (!emailNewlyCreated) {
                 throw error.unverifiedPrimaryEmailNewlyCreated();
+              } else {
+                throw error.unverifiedPrimaryEmailHasActiveSubscription();
```

Comment:
There can be two reasons now why an unverified account is not deleted, so I created a new `AppError` here.